app/models/pydantic/FolderModel.py

```diff
@@ -0,0 +1,31 @@
+"""
+This module provides the FolderPydanticModel class.
+"""
+from fastapi import HTTPException
+from typing import Optional, List
+
+from app.models.pydantic.AcademicYearModel import AcademicYearPydanticModel
+from app.models.pydantic.UEModel import PydanticUEModel
+from app.utils.enums.http_errors import CommonErrorMessages
+
+
+class PydanticFolderModel(AcademicYearPydanticModel):
+    name: str
+    children: Optional[List["PydanticFolderModel"]] = None
+    ue: Optional[PydanticUEModel] = None
+
+    def validate_children_or_ue (cls,value):
```

Comment:
Wtf is that ?
There is no doctring, if you use cls as first parameter, you should decorate this method with @classmethod.
What does it do ?
Children are always Nodes even the UE, you can maybe use a Composite pattern to implement it ?

Reply:
C'est pour qu'un dossier contienne soit une UE soit un autre dossier et pas les 2, et je l'ai fait avant que le NodeModel existe donc si ça te va pas je vais juste le supprimer vu que de toute façon il n'est pas utilisé